src/Pages/products/single-product-page.js

```diff
@@ -0,0 +1,119 @@
+import { useNavigate, useParams } from 'react-router-dom'
+import { useState } from 'react'
+import { Button } from '../../components/atomic'
+import { SmallLoader } from '../../components/composite'
+import { useCart } from '../../helpers/contexts/cart-context'
+import { useData } from '../../helpers/contexts/data-context'
+import './products-page.css'
+import {
+    useAddCartItem,
+    useAddWishItem,
+    useRemoveWishItem,
+} from '../../components/composite/Product-cards/Item-Hooks'
+import { useWishList } from '../../helpers/contexts/wishlist-context'
+export function ProductPage() {
```

Comment:
- well-structured code
- clean and easy to understand
- good use of custom hooks